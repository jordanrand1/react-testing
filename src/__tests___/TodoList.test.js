import TodoList from '../TodoList';
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

describe('<TodoList />', () => {
  
  describe('render', () => {
    let tree

    beforeEach( () => {
      let component = shallow(<TodoList/>)
      tree = toJson(component)
    })

    it ('matches a snapshot', () => {
      expect(tree).toMatchSnapshot()
    })

    it ('matches a snapshot with props', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('functionality', () => {
    let component

    beforeEach( () => {
      component = shallow(<TodoList />)
      const name = 'hello world'
      const input = component.find('input')
      input.simulate('focus')
      input.simulate(
        'change',
        { target: {name: 'name', value: name }}
      )
    })

    it('updates state on change', () => {
      expect(component.state('name')).toEqual(name)
    })

    it('submits the form', () => {
      const e = { preventDefault: jest.fn()}
      component.find('form').simulate('submit', e)
      expect(component.state('items')).toEqual([name])
    })
  })

})
