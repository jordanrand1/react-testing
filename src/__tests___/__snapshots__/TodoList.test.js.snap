// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> render matches a snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <label
      style={Object {}}
    >
      Add a Todo
    </label>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Add Item"
      required={true}
      value=""
    />
    <ul />
  </form>
</Fragment>
`;

exports[`<TodoList /> render matches a snapshot with props 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <label
      style={Object {}}
    >
      Add a Todo
    </label>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Add Item"
      required={true}
      value=""
    />
    <ul />
  </form>
</Fragment>
`;
